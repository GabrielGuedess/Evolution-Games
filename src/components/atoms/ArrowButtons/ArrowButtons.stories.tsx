import { Story, Meta } from '@storybook/react/types-6-0';

import { ArrowButtons } from './ArrowButtons';

export default {
  title: 'Atoms/ArrowButtons',
  component: ArrowButtons,
} as Meta;

export const Default: Story = () => <ArrowButtons />;
