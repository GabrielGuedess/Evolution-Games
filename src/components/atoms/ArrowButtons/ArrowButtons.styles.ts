import styled, { css } from 'styled-components';

type ButtonProps = {
  direction: 'left' | 'right';
};

const wrapperModifiers = {
  left: () => css`
    border-radius: 5px 0 0 5px;
  `,

  right: () => css`
    transform: scaleX(-1);
    border-radius: 5px 0 0 5px;
  `,
};

export const Wrapper = styled.div`
  min-width: 6.4rem;
`;

export const Button = styled.button<ButtonProps>`
  ${({ theme, direction }) => css`
    width: 3.2rem;
    height: 3.2rem;
    border: 1px solid ${theme.colors.secondary};
    color: ${theme.colors.secondary};
    transition: color 0.3s linear, border-color 0.3s linear;

    svg {
      width: 0.7rem;
      height: 1.2rem;
    }

    ${!!direction && wrapperModifiers[direction]}

    :hover,
    :focus {
      cursor: pointer;
      color: ${theme.colors.primary};
      border-color: ${theme.colors.primary};
    }
  `}
`;
