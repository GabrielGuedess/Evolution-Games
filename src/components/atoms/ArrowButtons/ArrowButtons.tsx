import * as S from './ArrowButtons.styles';

export type ArrowButtonsProps = {
  prevId?: string;
  nextId?: string;
};

export const ArrowButtons = ({ prevId, nextId }: ArrowButtonsProps) => (
  <S.Wrapper>
    <S.Button id={prevId} direction="left">
      <svg
        fill="none"
        aria-hidden="true"
        viewBox="0 0 8 13"
        preserveAspectRatio="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          fill="currentColor"
          d="M7.258 12.466a.937.937 0 0 0 0-1.325L2.618 6.5l4.64-4.64A.937.937 0 0 0 5.933.533L.629 5.837a.938.938 0 0 0 0 1.326l5.304 5.303a.937.937 0 0 0 1.325 0Z"
        />
      </svg>
    </S.Button>

    <S.Button id={nextId} direction="right">
      <svg
        fill="none"
        aria-hidden="true"
        viewBox="0 0 8 13"
        preserveAspectRatio="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          fill="currentColor"
          d="M7.258 12.466a.937.937 0 0 0 0-1.325L2.618 6.5l4.64-4.64A.937.937 0 0 0 5.933.533L.629 5.837a.938.938 0 0 0 0 1.326l5.304 5.303a.937.937 0 0 0 1.325 0Z"
        />
      </svg>
    </S.Button>
  </S.Wrapper>
);
