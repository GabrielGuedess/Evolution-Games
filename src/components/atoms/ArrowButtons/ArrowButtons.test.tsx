import { renderWithProviders } from 'utils/tests/helpers';

import { ArrowButtons } from './ArrowButtons';

describe('<ArrowButtons />', () => {
  it('should render the heading', () => {
    const { container } = renderWithProviders(
      <ArrowButtons nextId="swiper-next-id" prevId="swiper-prev-id" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
