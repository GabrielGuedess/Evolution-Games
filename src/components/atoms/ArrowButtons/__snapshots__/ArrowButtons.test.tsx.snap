// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<ArrowButtons /> should render the heading 1'] = `
.c0 {
  min-width: 6.4rem;
}

.c1 {
  width: 3.2rem;
  height: 3.2rem;
  border: 1px solid #717B8A;
  color: #717B8A;
  -webkit-transition: color 0.3s linear,border-color 0.3s linear;
  transition: color 0.3s linear,border-color 0.3s linear;
  border-radius: 5px 0 0 5px;
}

.c1 svg {
  width: 0.7rem;
  height: 1.2rem;
}

.c1:hover,
.c1:focus {
  cursor: pointer;
  color: #855aff;
  border-color: #855aff;
}

.c2 {
  width: 3.2rem;
  height: 3.2rem;
  border: 1px solid #717B8A;
  color: #717B8A;
  -webkit-transition: color 0.3s linear,border-color 0.3s linear;
  transition: color 0.3s linear,border-color 0.3s linear;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  border-radius: 5px 0 0 5px;
}

.c2 svg {
  width: 0.7rem;
  height: 1.2rem;
}

.c2:hover,
.c2:focus {
  cursor: pointer;
  color: #855aff;
  border-color: #855aff;
}

<div
  class="c0"
>
  <button
    class="c1"
    direction="left"
    id="swiper-prev-id"
  >
    <svg
      aria-hidden="true"
      fill="none"
      preserveAspectRatio="none"
      viewBox="0 0 8 13"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.258 12.466a.937.937 0 0 0 0-1.325L2.618 6.5l4.64-4.64A.937.937 0 0 0 5.933.533L.629 5.837a.938.938 0 0 0 0 1.326l5.304 5.303a.937.937 0 0 0 1.325 0Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <button
    class="c2"
    direction="right"
    id="swiper-next-id"
  >
    <svg
      aria-hidden="true"
      fill="none"
      preserveAspectRatio="none"
      viewBox="0 0 8 13"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.258 12.466a.937.937 0 0 0 0-1.325L2.618 6.5l4.64-4.64A.937.937 0 0 0 5.933.533L.629 5.837a.938.938 0 0 0 0 1.326l5.304 5.303a.937.937 0 0 0 1.325 0Z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
